internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestEmptyPush.java

```diff
@@ -0,0 +1,148 @@
+package com.linkedin.venice.endToEnd;
+
+import static com.linkedin.venice.kafka.TopicManager.DEFAULT_KAFKA_OPERATION_TIMEOUT_MS;
+import static com.linkedin.venice.utils.IntegrationTestPushUtils.defaultVPJProps;
+import static com.linkedin.venice.utils.IntegrationTestPushUtils.runVPJ;
+import static com.linkedin.venice.utils.IntegrationTestPushUtils.sendStreamingRecord;
+import static com.linkedin.venice.utils.TestWriteUtils.STRING_SCHEMA;
+import static com.linkedin.venice.utils.TestWriteUtils.getTempDataDirectory;
+import static com.linkedin.venice.utils.TestWriteUtils.writeEmptyAvroFileWithUserSchema;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import com.linkedin.venice.client.store.AvroGenericStoreClient;
+import com.linkedin.venice.client.store.ClientConfig;
+import com.linkedin.venice.client.store.ClientFactory;
+import com.linkedin.venice.compression.CompressionStrategy;
+import com.linkedin.venice.controllerapi.ControllerClient;
+import com.linkedin.venice.controllerapi.JobStatusQueryResponse;
+import com.linkedin.venice.controllerapi.UpdateStoreQueryParams;
+import com.linkedin.venice.controllerapi.VersionCreationResponse;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.integration.utils.ServiceFactory;
+import com.linkedin.venice.integration.utils.VeniceClusterCreateOptions;
+import com.linkedin.venice.integration.utils.VeniceClusterWrapper;
+import com.linkedin.venice.kafka.TopicManager;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.utils.IntegrationTestPushUtils;
+import com.linkedin.venice.utils.TestUtils;
+import com.linkedin.venice.utils.Utils;
+import java.io.File;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import org.apache.samza.system.SystemProducer;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class TestEmptyPush {
+  private VeniceClusterWrapper venice;
+
+  @BeforeClass(alwaysRun = true)
+  public void setUp() {
+    VeniceClusterCreateOptions options = new VeniceClusterCreateOptions.Builder().numberOfControllers(1)
+        .numberOfRouters(1)
+        .numberOfServers(2)
+        .sslToKafka(false)
+        .sslToStorageNodes(false)
+        .build();
+    venice = ServiceFactory.getVeniceCluster(options);
+  }
+
+  @AfterClass(alwaysRun = true)
+  public void tearDown() {
+    Utils.closeQuietlyWithErrorLogged(venice);
+  }
+
+  @Test
```

Comment:
Added a test timeout: 2min

I was trying to avoid adding too many integration tests to extend the CI running time.
I think the modified tests should be good enough to validate these scenarios:
1. First empty push with dict compression enabled to a hybrid store without any records in RT.
2. Writing some records to this hybrid store and validate them.
3. Second empty push with Gzip enabled and validate the records.
4. Third empty push with dict compression enabled, and the dict built this time will be different from 1), which would build some dummy dict.
5. Fourth empty push with dict compression enabled, and the dict built this time should be the same as the 3rd empty push.

Reply:
I don't think we should shy away from comprehensive testing... the tests run in parallel now so it's not as much a concern as before (as long as there isn't a single test class that runs super long...).

Reply:
Thanks Gaojie. This test covers all the scenarios I was checking about.